main/init/buildgen/src/mill/main/buildgen/BuildGenUtil.scala

```diff
@@ -46,7 +52,7 @@ object BuildGenUtil {
 
   }
 
-  def renderIrPom(value: IrPom): String = {
+  def renderIrPom(value: IrPom | Null): String = {
```

Comment:
Do these `| Null` types get checked by the compiler? Previously we didn't have the `| Null` and it seems we were sometimes passing `null` anyway

Reply:
Yes they do. I see you bumped the Scala version to 3 so I adopted [explicit nulls](https://docs.scala-lang.org/scala3/reference/experimental/explicit-nulls.html) to make the nullable types more explicit.